Dalamud/Plugin/Services/IGameInteropProvider.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Diagnostics;
+
+using Dalamud.Hooking;
+using Dalamud.Utility.Signatures;
+
+namespace Dalamud.Plugin.Services;
+
+/// <summary>
+/// Service responsible for the creation of hooks.
+/// </summary>
+public interface IGameInteropProvider
+{
+    /// <summary>
+    /// Available hooking backends.
+    /// </summary>
+    public enum HookBackend
+    {
+        /// <summary>
+        /// Choose the best backend automatically.
+        /// </summary>
+        Automatic,
+        
+        /// <summary>
+        /// Use Reloaded hooks.
+        /// </summary>
+        Reloaded,
+        
+        /// <summary>
+        /// Use MinHook.
+        /// You should never have to use this without talking to us first.
+        /// </summary>
+        MinHook,
+    }
+    
+    /// <summary>
+    /// Initialize <see cref="Hook{T}"/> members decorated with the <see cref="SignatureAttribute"/>.
+    /// Errors for fallible signatures will be logged.
+    /// </summary>
+    /// <param name="self">The object to initialize.</param>
+    public void InitializeFromAttributes(object self);
+    
+    /// <summary>
+    /// Creates a hook by rewriting import table address.
```

Comment:
Stray copy-paste from FromImport below?

Reply:
goat asked me what I would put here. I'm bad at this, but probably something along the lines of:
> Creates a hook by replacing the original address with an address pointing to a newly created jump to the detour.

Just thought I'd mention it here, since it kinda got burried in Discord. 🤷‍♂️🙂